src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/ModuleAutoTool.kt

```diff
@@ -32,13 +32,19 @@ import net.ccbluex.liquidbounce.features.module.Module
 
 object ModuleAutoTool : Module("AutoTool", Category.WORLD) {
 
-    val handler = handler<BlockAttackEvent> { event ->
+    val handler = handler<BlockBreakingProgressEvent> { event ->
         val blockState = world.getBlockState(event.pos)
+        var bestSlot: Int? = null
+        var bestSpeed = 1f
 
-        val bestSlot = (0..8).maxByOrNull { i ->
-            val item = player.inventory.getStack(i) ?: return@maxByOrNull -1.0f
+        for (i in 0..8) {
+            val item = player.inventory.getStack(i)
+            val speed = item.getMiningSpeedMultiplier(blockState)
 
-            item.getMiningSpeedMultiplier(blockState)
+            if (speed > bestSpeed) {
+                bestSpeed = speed
+                bestSlot = i
+            }
```

Comment:
Why did you replace you replace this function by something more complicated?

Reply:
Because the maxByOrNull function checks if value `a` is greater than or `equal` to value b, which in game that leads to the module choosing slot number 1 everytime when the item mining speed of tools are the same as the character's hand mining speed.